torch/_subclasses/fake_impls.py

```diff
@@ -171,6 +171,8 @@ def constructors(fake_mode, func, *args, **kwargs):
     out_device = new_kwargs.pop("device", None)
     out_device = out_device if out_device is not None else default_device
     new_kwargs["device"] = torch.device("meta")
+    if "s" in new_kwargs and isinstance(new_kwargs["s"], torch.SymFloat):
+        new_kwargs["dtype"] = torch.float64
```

Comment:
even we set this argument, it can't be passed to the lowering stage. I add it just want to keep the fx graph nodes has the correct annotation with the later lowering. Do you think we should remove it?

Reply:
I don't know what your question is asking, but this code suuuure looks suspicious

Reply:
The following line `r = func(*args, **new_kwargs)` will call [scalar_tensor](https://github.com/pytorch/pytorch/blob/main/torch/_meta_registrations.py#L5766) to create a FakeTensor for the `'s':zuf0` in the `new_kwargs`. However, there is no data type information passed to the scalar_tensor function, so, it will create a faketensor with the default float32, even the input tensor is float64 in our testing example. 
So, I add the check for `'s':zuf0` here to pass its dtype.  I found the `dtype` added can be passed to the fx graph node interpret. But it can't be passed to the graph lowering. because when we do the graph lowering for the `scalar_tensor` node, it uses the function [tensor](https://github.com/pytorch/pytorch/blob/ed1b46ddb10760d37ea61393c757fdc70a79b9df/torch/_inductor/lowering.py#L2498) in lowering.py to create the tensor. But the arguments passed only incudes `{'device': device(type='cpu'), 'pin_memory': False}`. So, I have to update the kwargs in the above [torch/_dynamo/variables/tensor.py](https://github.com/pytorch/pytorch/pull/131775/files#diff-3c5ada49e5ae30dfde224c1ae8eabd809c4337f0a1a99528feee91d0560779f4R1065-R1067) file.

Reply:
I feel like the best place should be before this function and directly update the `kwargs` passed in this function.

Reply:
Moved to `scalar_tensor` in torch/_meta_registrations.py https://github.com/pytorch/pytorch/pull/131775/commits/1251f4bca6bf108f1872653f1db73ca4c1a1778e